src/app/store/reducer/cart.reducer.ts

```diff
@@ -0,0 +1,49 @@
+import { ICartItem } from 'src/app/model/cart-item.model';
+import { CartActions, ECartActions } from '../action/cart.action';
+import { ICartState, initialCartState } from '../state/cart.state';
+
+export const cartReducer = (
+  state = initialCartState,
+  action: CartActions
+): ICartState => {
+  switch (action.type) {
+    case ECartActions.AddCartItemSuccess: {
+      let item = state.items.find((item) => item.productId == action.payload);
```

Comment:
Here this reducer case has a lot of logic. Complex logic usually goes in the Effects, the reducer stays as a pure function with no state. Also try to use const (e.g: for the "item" variable in this case) most of the time.